File: backend/app/schemas/k8s_objects.py
from datetime import datetime

from pydantic import BaseModel


class StatefulSetInfo(BaseModel):
    name: str
    namespace: str
    replicas: int
    ready_replicas: int
    service_name: str | None
    images: list[str]
    created_at: datetime | None


class DaemonSetInfo(BaseModel):
    name: str
    namespace: str
    desired: int
    current: int
    ready: int
    available: int
    images: list[str]
    created_at: datetime | None

File: frontend/src/types/k8s_objects.ts
export interface StatefulSetInfo {
  name: string;
  namespace: string;
  replicas: number;
  ready_replicas: number;
  service_name: string | null;
  images: string[];
  created_at: string | null;
}

export interface DaemonSetInfo {
  name: string;
  namespace: string;
  desired: number;
  current: number;
  ready: number;
  available: number;
  images: string[];
  created_at: string | null;
}
